tests/helpers/common/validator.go

```diff
@@ -205,41 +204,71 @@ func BuildValidator(ctx context.Context, runenv *runtime.RunEnv, initCtx *run.In
 		return nil, err
 	}
 
-	var persPeers []string
-	for i := 0; i < runenv.IntParam("persistent-peers"); i++ {
+	var peers []appkit.ValidatorNode
+	for i := 0; i < runenv.IntParam("validator"); i++ {
 		select {
 		case err = <-sub.Done():
 			if err != nil {
 				return nil, err
 			}
 		case val := <-valCh:
-			runenv.RecordMessage("Validator Received: %s, %s", val.IP, val.PubKey)
 			if !val.IP.Equal(ip) {
-				persPeers = append(persPeers, fmt.Sprintf("%s@%s", val.PubKey, val.IP.To4().String()))
-			}
-
-			err = appkit.AddPersistentPeers(configPath, persPeers)
-			if err != nil {
-				return nil, err
+				peers = append(peers, *val)
 			}
 		}
 	}
+	runenv.RecordMessage("Validator Received is equal to: %d", len(peers))
+
+	randomizer := tmrand.Intn(runenv.IntParam("validator"))
+	runenv.RecordMessage("Randomized number is equal to: %d", randomizer)
+	peersRange := runenv.IntParam("persistent-peers")
```

Comment:
FYI, this should be 
`peersRange := runenv.IntParam("persistent-peers") -1`
or 
`peersRange := math.MinInt(runenv.IntParam("persistent-peers"), len(peers))`

But maybe just use the `peers` array for `peersRange`

Reply:
The motivation is mainly to get any randomised peers from a set of Y for a range of X (e.g. give me any 10 rand peers from a set of 99 in a Y = 100 val set, X = 10 peers)

First proposal is just downgrading peers to the address book of peers for each validator to 9 
For the second one: Could you please elaborate on the outcome of it? 

Still it's an awesome idea. I would much consider not a constant number of range but a randomised
Smt like this:

```go
// 0 peers is very bad to start, but might be cool to see how seed nodes works 
peersRange := rand.Intn(runenv.IntParam("persistent-peers")) + 1
```

P.S. `persistent-peers` is a horrible legacy naming that I will change to smt like `addrbook-peers`

Appreciate the feedback